qiskit/algorithms/evolvers/variational/solvers/var_qte_linear_solver.py

```diff
@@ -30,29 +32,63 @@ class VarQTELinearSolver:
 
     def __init__(
         self,
-        metric_tensor: OperatorBase,
+        ansatz: Union[StateFn, QuantumCircuit],
+        qfi: QFI,
+        gradient_params: List[Parameter],
         evolution_grad: OperatorBase,
+        t_param=None,
         lse_solver: Callable[[np.ndarray, np.ndarray], np.ndarray] = np.linalg.lstsq,
-        circuit_sampler: Optional[CircuitSampler] = None,
+        quantum_instance: Optional[QuantumInstance] = None,
         imag_part_tol: float = 1e-7,
     ) -> None:
         """
         Args:
-            metric_tensor: A parametrized operator that represents the left-hand side of an ODE.
+            ansatz: Quantum state in the form of a parametrized quantum circuit.
+            qfi: A Quantum Fisher Information instance used to calculate a metric tensor for the
+                left-hand side of an ODE.
+            gradient_params: List of parameters with respect to which gradients should be computed.
             evolution_grad: A parametrized operator that represents the right-hand side of an ODE.
+            t_param: Time parameter in case of a time-dependent Hamiltonian.
             lse_solver: Linear system of equations solver that follows a NumPy
                 ``np.linalg.lstsq`` interface.
-            circuit_sampler: Samples circuits using an underlying backend.
+            quantum_instance: Backend used to evaluate the quantum circuit outputs. If ``None``
+                provided, everything will be evaluated based on matrix multiplication (which is
+                slow).
             imag_part_tol: Allowed value of an imaginary part that can be neglected if no
                 imaginary part is expected.
         """
-
-        self._metric_tensor = metric_tensor
+        self._ansatz = ansatz
+        self._qfi = qfi
```

Comment:
Do we need this private class variable, since we already construct the QFI callable?